articles/iot-hub/iot-hub-devguide-messages-d2c.md

```diff
@@ -119,7 +119,7 @@ You can enable/disable the fallback route in the Azure portal->Message Routing b
 
 ## Non-telemetry events
 
-In addition to device telemetry, message routing also enables sending device twin change events, device lifecycle events, digital twin change events, and device connection state events. For example, if a route is created with data source set to **device twin change events**, IoT Hub sends messages to the endpoint that contain the change in the device twin. Similarly, if a route is created with data source set to **device lifecycle events**, IoT Hub sends a message indicating whether the device was deleted or created. As part of [Azure IoT Plug and Play](../iot-develop/overview-iot-plug-and-play.md), a developer can create routes with data source set to **digital twin change events** and IoT Hub sends messages whenever a digital twin property is set or changed, a digital twin is replaced, or when a change event happens for the underlying device twin. Finally, if a route is created with data source set to **device connection state events**, IoT Hub sends a message indicating whether the device was connected or disconnected.
+In addition to device telemetry, message routing also enables sending device twin change events, device lifecycle events, digital twin change events, and device connection state events. For example, if a route is created with data source set to **device twin change events**, IoT Hub sends messages to the endpoint that contain the change in the device twin. Similarly, if a route is created with data source set to **device lifecycle events**, IoT Hub sends a message indicating whether the device or module was deleted or created. More details on device lifecycle events can be found [here](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#device-and-module-lifecycle-notifications). As part of [Azure IoT Plug and Play](../iot-develop/overview-iot-plug-and-play.md), a developer can create routes with data source set to **digital twin change events** and IoT Hub sends messages whenever a digital twin property is set or changed, a digital twin is replaced, or when a change event happens for the underlying device twin. Finally, if a route is created with data source set to **device connection state events**, IoT Hub sends a message indicating whether the device was connected or disconnected.
```

Comment:
Not your change - but I detest using the work "Digital Twin" outside of ADT.  It's _so_ confusing.

Reply:
Use the relative path with .md for docs links

./iot-hub-devguide-identity-registry.md#device-and-module-lifecycle-notifications

Reply:
Unfortunately, "Digital Twins" is what the feature is actually called. It shows up as a route source in the portal too.  Resolved the link - I didn't know how to put a section partway down the page. :) now I do with the #.